app/queue/egress.py

```diff
@@ -48,11 +48,19 @@ def write_event(self, event, key=None):
 
 
 @metrics.egress_event_serialization_time.time()
-def build_event(event_type, host, metadata):
+def build_event(event_type, host, platform_metadata, metadata=None):
     if event_type in ("created", "updated"):
         return (
             HostEvent(strict=True)
-            .dumps({"type": event_type, "host": host, "platform_metadata": metadata, "timestamp": datetime.utcnow()})
+            .dumps(
+                {
+                    "type": event_type,
+                    "host": host,
+                    "platform_metadata": platform_metadata,
+                    "metadata": metadata,
```

Comment:
The only metadata that is put in here is request ID. The existing delete event has it directly in the object root. Since the original Egress event didn’t have this field, we are free to put it there too, aren’t we? Or is there a contract saying that the request ID is expected under the _metadata_ subfield.

```suggestion
                    "request_id": request_id,
```

Reply:
Jozef addressed this in his comment below. Marking as resolved.